src/pheromones/src/GetNeighborResponseDTO.cpp

```diff
@@ -0,0 +1,35 @@
+#include "GetNeighborResponseDTO.h"
+
+GetNeighborResponseDTO::GetNeighborResponseDTO(const GetNeighborResponse resp) :
+    m_neighborId(resp.neighbor_id) {
+    if (resp.has_position) {
+        m_pos = NeighborPositionDTO(resp.position);
+    } else {
+        m_pos = {};
+    }
+}
+
+GetNeighborResponseDTO::GetNeighborResponseDTO(uint16_t neighborId,
+                                               const std::optional<NeighborPositionDTO>& pos) :
+    m_neighborId(neighborId), m_pos(pos) {}
+
+uint16_t GetNeighborResponseDTO::getNeighborId() const { return m_neighborId; }
+
+const std::optional<NeighborPositionDTO>& GetNeighborResponseDTO::getNeighborPosition() const {
+    return m_pos;
+}
+
+void GetNeighborResponseDTO::setNeighborId(uint16_t id) { m_neighborId = id; }
+
+void GetNeighborResponseDTO::setNeighborPosition(const std::optional<NeighborPositionDTO>& pos) {
+    m_pos = pos;
+}
+
+bool GetNeighborResponseDTO::serialize(GetNeighborResponse& resp) const {
+    resp.neighbor_id = m_neighborId;
+    resp.has_position = m_pos.has_value();
+    if (m_pos) {
```

Comment:
if no m_pos, isn't because m_pos is null?
If so, m_pos.has_value() might create a nullptr exception

Reply:
It's not a pointer here, it's an std::optional, which implement boo() (what is used in an if conditional). So you can just check with an if it has a value. has_value is a method from std::optional. 

doing 

```cpp
std::optional<int> optionalVar = {}; // empty
if(optionalVar) {
// will not  be called
}
```

is the same as 

```cpp
std::optional<int> optionalVar = {}; // empty
if(optionalVar.has_value()) {
// will not  be called
}
```

just the second one is more verbose for no reason

Reply:
nice, good to know!